apps/dapp/src/components/Pages/Core/DappPages/Dashboard/Table/TxnHistoryTable.tsx

```diff
@@ -1,72 +1,164 @@
+import { TableRow, TxType, TxnDataTable } from './TxnDataTable';
+import { PaginationControl } from './PaginationControl';
+import { RowFilter, useTxHistory, useTxHistoryAvailableRows } from '../hooks/use-dashboardv2-txHistory';
+import { useDebouncedCallback } from 'use-debounce';
+import { StrategyKey } from '../hooks/use-dashboardv2-metrics';
+import { SelectTempleDaoOptions } from 'components/InputSelect/InputSelect';
 
 type Props = {
   dashboardType: DashboardType;
-  filter: TxHistoryFilterType;
+  txFilter: TxHistoryFilterType;
 };
 
-const TxnHistoryTable = ({ dashboardType, filter }: Props) => {
+export enum TableHeaders {
+  Date = 'Date',
+  Type = 'Type',
+  Strategy = 'Strategy',
+  Token = 'Token',
+  Amount = 'Amount',
+  TxHash = 'Tx Hash',
+}
+
+enum DebtToken {
+  DAI = 'DAI',
+  TEMPLE = 'TEMPLE',
+}
+
+export type TxHistoryTableHeader = {
+  name: TableHeaders;
+  orderDesc?: boolean;
+  filter?: (event: HTMLInputElement) => void;
+  dropdownOptions?: SelectTempleDaoOptions;
+}
+
+const TxnHistoryTable = ({ dashboardType, txFilter }: Props) => {
   const [currentPage, setCurrentPage] = useState(1);
   const [rowsPerPage, setRowsPerPage] = useState(10);
+  const [rowFilter, setRowFilter] = useState<RowFilter>({});
+
+  const [tableHeaders, setTableHeaders] = useState<TxHistoryTableHeader[]>([
+    {
+      name: TableHeaders.Date,
+      orderDesc: true,
+    },
+    {
+      name: TableHeaders.Type,
+      orderDesc: undefined,
+      filter: useDebouncedCallback(async (event: HTMLInputElement) => {
+        setRowFilter(s => ({...s, type: event.value}));
+      }, 200),
+      // TODO: get the values programatically from subpgrah
+      dropdownOptions: [
+        { label:'All', value: '' },
+        { label: TxType.Borrow, value: TxType.Borrow },
+        { label: TxType.Repay, value: TxType.Repay },
+      ]
+    },
+    {
+      name: TableHeaders.Strategy,
+      orderDesc: undefined,
+      filter: useDebouncedCallback(async (event: HTMLInputElement) => {
+        setRowFilter(s => ({...s, strategy: event.value}));
+      }, 200),
+      // TODO: get the values programatically from subpgrah
+      dropdownOptions: [
+        { label:'All', value: '' },
+        { label: StrategyKey.RAMOS, value: StrategyKey.RAMOS },
+        { label: StrategyKey.TLC, value: StrategyKey.TLC },
+        { label: StrategyKey.TEMPLEBASE, value: StrategyKey.TEMPLEBASE },
+        { label: StrategyKey.DSRBASE, value: StrategyKey.DSRBASE },
+      ]
+    },
+    {
+      name: TableHeaders.Token,
+      orderDesc: undefined,
+      filter: useDebouncedCallback(async (event: HTMLInputElement) => {
+        setRowFilter(s => ({...s, token: event.value}));
+      }, 200),
+      // TODO: get the values programatically from subpgrah
+      dropdownOptions: [
+        { label:'All', value: '' },
+        { label: DebtToken.DAI, value: DebtToken.DAI },
+        { label: DebtToken.TEMPLE, value: DebtToken.TEMPLE },
+      ]
```

Comment:
is there a query from subgraph to get this dropdown options programmatically @medariox?

Reply:
> is there a query from subgraph to get this dropdown options programmatically @medariox?

No, there is no support for that in GraphQL directly. You'd have to query the entities and filter out the duplicates with a set for example.

Reply:
but we would get again to the issue of 1000 max records per query, if we didn't get all the options in the first 1000 records, it'll be tricky to get all the million of records we could have, then filter out duplicates in the frontend, would prob just leave the dropdowns hardcoded for now